GLE_solver_v4.py

```
@@ -0,0 +1,331 @@
+    
+    print(f"Finding s_max for h_end = {h_end:.3f}, theta_end = {theta_end*180/np.pi:.1f}°")
+    
+    # Check if bounds bracket the root
+    f_min = solve_gle_for_s_max(s_min)
+    f_max = solve_gle_for_s_max(s_max_upper)
+    
+    print(f"  At s_min = {s_min:.2e}: h - h_end = {f_min:.3e}")
+    print(f"  At s_max = {s_max_upper:.2e}: h - h_end = {f_max:.3e}")
+    
+    if f_min * f_max > 0:
+        print(f"  Warning: Root may not be bracketed. Expanding search range...")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)